src/utils/components/CPUMemoryModal/CpuMemoryModal.tsx

```diff
@@ -0,0 +1,200 @@
+import * as React from 'react';
+import produce from 'immer';
+
+import VirtualMachineModel from '@kubevirt-ui/kubevirt-api/console/models/VirtualMachineModel';
+import { V1VirtualMachine } from '@kubevirt-ui/kubevirt-api/kubevirt';
+import { useKubevirtTranslation } from '@kubevirt-utils/hooks/useKubevirtTranslation';
+import { k8sUpdate } from '@openshift-console/dynamic-plugin-sdk';
+import {
+  Alert,
+  Button,
+  ButtonVariant,
+  Dropdown,
+  DropdownItem,
+  DropdownToggle,
+  Modal,
+  ModalVariant,
+  NumberInput,
+  Title,
+  TitleSizes,
+} from '@patternfly/react-core';
+
+import useTemplateDefaultCpuMemory from './hooks/useTemplateDefaultCpuMemory';
+import { getCPUCount, getMemorySize, memorySizesTypes } from './utils/CpuMemoryUtils';
+
+import './cpu-memory-modal.scss';
+
+type CPUMemoryModalProps = {
+  vm: V1VirtualMachine;
+  isOpen: boolean;
+  onClose: () => void;
+};
+
+const CPUMemoryModal: React.FC<CPUMemoryModalProps> = ({ vm, isOpen, onClose }) => {
```

Comment:
we need to add an `onSubmit` prop

`onSubmit: (updatedVM: V1VirtualMachine) => Promise<void>`